docs/compiling.rst

```diff
@@ -1,6 +1,38 @@
+.. _compiling:
+
 Build systems
 #############
 
+Building manually
+=================
+
+pybind11 is a header-only-library, hence it is not necessary to link against
+any special libraries (other than Python itself).
+
+On Linux, the above example can be compiled using the following command:
+
+.. code-block:: bash
+
+    $ c++ -O3 -Wall -shared -std=c++11 -fPIC `python -m pybind11 --includes` example.cpp -o example`python3-config --extension-suffix`
+
+Note that on Python 2.7.x ``python-config`` has to be used instead of
+``python3-config`` in the command above. Besides, ``--extension-suffix``
+option may or may not be available, depending on the distribution; in the latter
+case, the module extension can be manually set to ``.so``.
```

Comment:
I think this whole paragraph is slightly obsolete by the #921 changes (since the only `python3-config` is now the extension suffix one).  Perhaps just simplify it to:

```
Note that Python 2.7 modules don't use a special suffix, so you should simply use `example.so` instead of ``example`python3-config --extension-suffix```.

Reply:
Where is the link to the Python library in this command? I need something like:

```bash
$ c++ -O3 -Wall -shared -std=c++11 -fPIC `python3 -m pybind11 --includes` `python3-config --libs` example.cpp -o example`python3-config --extension-suffix`
```

I'm also on macOS and didn't been the ``--undefined dynamic_lookup`` part, at least for the built-in Python 2 and the sample in #1041.

Reply:
On Linux, it's better to (intentionally) not link against libpython. The symbols will be resolved when the extension library is loaded into a Python binary. This is preferable because you might have several different installations of a given Python version (e.g. the system-provided Python, and one that ships with a piece of commercial software). In this way, the plugin will work with both versions, instead of possibly importing a second Python library that will lead to a segfault.

Reply:
Ahh, so I traded the undefined for the link. Nevermind, then! Thanks.